### ISSUE TITLE
Cannot compile, execvp missing

### ISSUE THREAD

**Issue by sasdallas** [lordmilko/i686-elf-tools#13]
Hello! Sorry for asking another question, but I'm having an issue. It's with a different host, not Docker container.

I have built the i686-elf-tools using the script and installed them to `/usr/local`. note: this has also been tried without copying to `/usr/local`. I have tried to execute this command: `i686-elf-gcc -c kernel.c -o kernel.o -std=gnu99 -ffreestanding -O2 -Wall -Wextra`, but I get an error: 
`i686-elf-gcc: error trying to exec 'cc1': execvp: Permission denied`

How do I resolve this?
Thanks!

Linux Distro: Kali Linux 2021.3

**Comment by lordmilko** [comment]
Hi @sasdallas,

I tested building i686-elf-tools under Kali 2021.4. My build was performed as `root`. Two issues I had during the build were that `python` is not a valid package in Kali (solved by installing `python-is-python3` (I don't know if maybe it's meant to be python2 or python3 though) and removing `python` from the build script) and GDB failed to build due to some issues with some Python build dependencies (solved by simply doing `./i686-elf-tools.sh binutils gcc` since I don't need GDB to investigate your issue)

After the build completed I did

```sh
cd ~/build-i686-elf/linux
cp -r output/* /usr/local/
```

Then I created a file `kernel.c` with the text

```c++
int main()
{
    return 0;
}
```
and successfully executed the command you used

```sh
i686-elf-gcc -c kernel.c -o kernel.o -std=gnu99 -ffreestanding -O2 -Wall -Wextra
```
Given your issue appears to have to do with permissions, I would ask
* What account did you use to build i686-elf-tools?
* What account did you use to copy it to /usr/local?
* What version of GCC did you build?
* Assuming the default of 7.1.0, what is the output of ` ls -l /usr/local/libexec/gcc/i686-elf/7.1.0/cc1`
* What account are you trying to run `i686-elf-gcc` as?
